frontend/src/FullPageRoutes/VideoListPage.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { request } from '../utils/fetch';
+
+type Video = {
+    id: number;
+    title: string;
+    video_file: string;
+};
+
+export default function VideoListPage() {
+    const [videos, setVideos] = React.useState<Video[]>([]);
+
+    React.useEffect(() => {
+        request<Video[]>('/api/video/', 'get', true, false).then((res) => {
+            setVideos(res.parsedBody);
+        });
+    });
+
+    return (
+        <ul>
+            {videos.map(({ id, title, video_file }) => {
+                return (
+                    <li key={'video' + id}>
+                        {title}
+                        <video width={500} height={500} controls>
+                            <source type='video/mp4' src={video_file} />

```

Comment:
I see the appeal of having a data URL in this scenario. Ultimately, the file serving probably shouldn't even be handled by the REST API backend and instead by a proper web server.

In the meantime, instead of sending the video file back, you could send the location of the file and use that as the source of the video. Django should allow you to specify the URL to the media in debug mode. Something like `localhost:8000/media/<video_file>`. Of course when you send the file location back, you are really specifying the whole URL so this code here would be largely untouched.